src/proposal/mod.rs

```diff
@@ -0,0 +1,151 @@
+pub mod repost;
+pub mod timeline;
+
+use std::collections::HashSet;
+
+use self::timeline::TimelineStatus;
+
+use crate::notify::get_text_mentions;
+use crate::str_serializers::*;
+use crate::{Balance, SponsorshipToken};
+
+use near_sdk::borsh::{BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::Base58CryptoHash;
+use near_sdk::schemars::JsonSchema;
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::{AccountId, BlockHeight, Timestamp};
+
+pub type ProposalId = u64;
```

Comment:
Let's either use u32 or near_sdk::json_type::U64, so we don't cause troubles for JS consumers that can only handle number values in JSON up to 2**53 (f64)